### PR TITLE
Update the header

### PR DESCRIPTION
**Pull request by 36degrees**
Updates the header to:

- use the new focus style for the menu button
- use most of the header markup and CSS from GOV.UK Frontend, with some targeted overrides where required
- align the search within the header, align the text within the search input correctly, and update its focus state

<details>
<summary>Screenshots</summary>
Screen Shot 2019-07-18 at 12 13 06
Screen Shot 2019-07-18 at 12 13 09
Screen Shot 2019-07-18 at 12 13 15

localhost_3000_components_back-link_
localhost_3000_components_(iPhone X)
localhost_3000_components_(iPhone X) (1)
</details>

This covers #961 and part of the work to remove baseline adjustments from the Design System (#957)

### CODE REVIEW

File: src/stylesheets/components/_header.scss [hunk 1]
```diff
@@ -135,25 +55,17 @@
       }
     }
 
+    &:active:not(:hover),
+    &:focus:not(:hover) {
+      &::after {
+        border-color: $govuk-focus-text-colour;
```
Comment (nickcolley) [inline src/stylesheets/components/_header.scss]: If you use `currentColor` I think you can simplify this to avoid so many selectors.

Comment (nickcolley) [reply src/stylesheets/components/_header.scss]: If you press 'space' on the button you'll see this isnt quite right, whereas currentColor avoids having to figure this out.

Comment (36degrees) [reply src/stylesheets/components/_header.scss]: @aliuk2012 ^

**Comment by 36degrees** [comment]
@nickcolley nice work on the menu button, though you now seem to get flashes of green background when you click it?

In IE11, it also goes green when you hover it?

Screenshot 2019-07-19 at 17 02 17BST

**Comment by nickcolley** [comment]
I can't reproduce the flashing of green can you let me know what you see please?

IE11 is a weird one, will have to investigate that, it seems the CSS is applied but doesn't work? I'll update so it's always `transparent`

**Comment by 36degrees** [comment]
> I can't reproduce the flashing of green can you let me know what you see please?

Now that I look again with fresh eyes, it _might_ just be Chrome's touch highlight?

flash

**Comment by nickcolley** [comment]
If we were to change that I think we could think about it in this issue: https://github.com/alphagov/govuk-frontend/issues/1205

**Comment by 36degrees** [comment]
LGTM, thanks @nickcolley 👍

**Pull request merged**